client/json-rpc/src/async_client/client.rs

```diff
@@ -475,6 +552,50 @@ impl<R: RetryStrategy> Client<R> {
         resp.json().await.map_err(Error::InvalidHTTPResponse)
     }
 
+    async fn multi_send_json_request<T: for<'de> serde::Deserialize<'de>>(
+        &self,
+        req: serde_json::Value,
+    ) -> Result<Vec<T>, Error> {
+        // Choose `num_parallel_requests` from `self.server_urls` to send requests to
+        let chosen_urls: Vec<reqwest::Url> = self
+            .server_urls
+            .choose_multiple(&mut rand::thread_rng(), self.num_parallel_requests)
+            .cloned()
+            .collect();
+        let futures = chosen_urls.iter().map(|url| {
+            let url = url.clone();
+            let req = req.clone();
+            async move {
+                match timeout(
+                    defaults::TIMEOUT,
+                    self.single_json_request(url.clone(), req),
+                )
+                .await
+                {
+                    Err(_) => Err(Error::ResponseTimeout(format!(
```

Comment:
I'm thinking about the case where one of the servers is down, we'd constantly be hitting this timeout. Should we consider caching the bad servers or something like that to avoid this?

Reply:
It's a good point. Depends on what down means. If the server is completely down, we should be hitting a connection error and returning immediately, this would really happen if the server is up, but really slow to respond.

The idea of caching bad servers sounds good, but it's beyond the scope of this PR.